WordPress/Jetpack/Classes/NUX/New Landing Screen/Views/InfiniteScrollerView.swift

```diff
@@ -0,0 +1,114 @@
+import Foundation
+import UIKit
+
+/// A scroll view that produces an infinite scrolling effect.
+///
+/// Set the `scrollerDelegate` property to externally alter the scrolling rate speed and direction.
+///
+/// By default user interaction with the scroll view is disabled, but it can be enabled if desired.
+///
+/// Currently the view only supports vertical scrolling, but could easily be updated to support horizontal.
+/// This class also currently makes the assumption that the view generated by the `viewBuilder` closure will be taller
+/// than its frame.
+class InfiniteScrollerView: UIScrollView {
+    /// Delegate to poll for the speed and direction of the scrolling offset. Polled at the default rate of `CADisplayLink`.
+    var scrollerDelegate: InfiniteScrollerViewDelegate?
+
+    private var displayLink: CADisplayLink?
+
+    /// Closure used to repeat a view to populate a `UIStackView` for scrolling.
+    /// Provides the index of the view in the `UIStackView`.
+    private var viewBuilder: ((Int) -> UIView)?
+    private var stackView: UIStackView?
+
+    /// Returns the height of the first view in the stack view, if one exists. Used to compute reset points for vertical scrolling.
+    private var singleViewHeight: CGFloat? {
+        guard let stackView = stackView, !stackView.arrangedSubviews.isEmpty else {
+            return nil
+        }
+
+        return stackView.arrangedSubviews[0].frame.size.height
+    }
+
+    /// Initializes a view that produces an infinite scrolling effect.
+    /// - Parameter viewBuilder: Closure that generates a `UIView`. Repeatedly called to fill a `UIStackView` that will be scrolled.
+    convenience init(_ viewBuilder: @escaping ((Int) -> UIView)) {
+        self.init()
+        self.viewBuilder = viewBuilder
+        displayLink = CADisplayLink(target: self, selector: #selector(step))
+
+        setupScrollView()
+        let stackView = setupStackView()
+        addSubview(stackView)
+
+        stackView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        stackView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        stackView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        stackView.widthAnchor.constraint(equalTo: widthAnchor).isActive = true
+
+        self.stackView = stackView
+        addViewsToStackView()
+
+        displayLink?.add(to: .current, forMode: .default)
+    }
+
+    private func addViewsToStackView() {
+        guard let viewBuilder = viewBuilder else { return }
```

Comment:
we might want to stick to the [style guide](https://github.com/wordpress-mobile/swift-style-guide) and put the return and the closing brace on different lines.